packages/babel-helper-compilation-targets/src/index.js

```diff
@@ -184,6 +184,9 @@ export default function getTargets(
     targets.browsers = Object.keys(supportsESModules)
       .map(browser => `${browser} ${supportsESModules[browser]}`)
       .join(", ");
+  } else {
+    // remove falsy esmodules to fix `hasTargets` below
+    delete targets.esmodules;
```

Comment:
Should we also delete it if it is truthy?

Reply:
User may invoke this method twice or more, keep the truthy value is safer for this case

```js
const getTargets = require( '@babel/helper-compilation-targets' ).default
const targets = { esmodules: true }
const result1 = getTargets( targets )
const result2 = getTargets( targets )
// result1 should be equal to result2 here
```

Reply:
Wouldn't they still be the same?

After the first call, targets has a `browsers` property which makes its behaviour identical to `esmodules: true`.

Reply:
Oops, sorry I didn't realize targets.browsers have been assigned. Updated